tools/dds/dds-config/rs-dds-config.cpp

```diff
@@ -0,0 +1,386 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include <librealsense2/rs.hpp>
+
+#include "eth-config.h"
+
+#include <tclap/CmdLine.h>
+#include <tclap/ValueArg.h>
+
+#include <rsutils/os/special-folder.h>
+#include <rsutils/json.h>
+#include <rsutils/json-config.h>
+#include <rsutils/string/from.h>
+
+#include <iostream>
+#include <thread>
+#include <set>
+
+using namespace TCLAP;
+using rsutils::json;
+using rsutils::type::ip_address;
+
+
+static json load_settings( json const & local_settings )
+{
+    // Load the realsense configuration file settings
+    std::string const filename = rsutils::os::get_special_folder( rsutils::os::special_folder::app_data ) + RS2_CONFIG_FILENAME;
+    auto config = rsutils::json_config::load_from_file( filename );
+
+    // Take just the 'context' part
+    config = rsutils::json_config::load_settings( config, "context", "config-file" );
+
+    // Patch the given local settings into the configuration
+    config.override( local_settings, "local settings" );
+
+    return config;
+}
+
+
+uint32_t const GET_ETH_CONFIG = 0xBB;
+uint32_t const SET_ETH_CONFIG = 0xBA;
+char const * const HWM_FMT = "{4} {04}({4i},{4i},{4i},{4i}) {repeat:}{1}{:}";
+
+
+#define LOG_DEBUG( ... )                                                                                               \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        std::ostringstream os__;                                                                                       \
+        os__ << __VA_ARGS__;                                                                                           \
+        rs2_log( RS2_LOG_SEVERITY_DEBUG, os__.str().c_str(), nullptr );                                                \
+    }                                                                                                                  \
+    while( false )
+bool g_quiet = false;
+#define INFO( ... )                                                                                                    \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        if( ! g_quiet )                                                                                                \
+            std::cout << "-I- " << __VA_ARGS__ << std::endl;                                                           \
+    }                                                                                                                  \
+    while( false )
+
+
+struct indent
+{
```

Comment:
Does this compile in all environments? member and input variable have same name and might hide one another.

Reply:
Yes, AFAIK. But I'll change anyway.